pkg/hooks/loader.go

```diff
@@ -225,14 +248,67 @@ func (h *Hook) GetConfigMocks() ([]*models.Mock, error) {
 
 func (h *Hook) DeleteTcsMock(mock *models.Mock) bool {
 	isDeleted := h.tcsMocks.delete(mock.TestModeInfo)
+	if isDeleted {
+		h.persistentMatchedMocks[mock.Name] = true
+	}
 	return isDeleted
 }
 
 func (h *Hook) DeleteConfigMock(mock *models.Mock) bool {
 	isDeleted := h.configMocks.delete(mock.TestModeInfo)
+	if isDeleted {
+		h.persistentMatchedMocks[mock.Name] = false
+	}
 	return isDeleted
 }
```

Comment:
incase if a mock is matched you're setting it to false , why?

Reply:
https://github.com/keploy/keploy/pull/1573#discussion_r1497570055